### PR TITLE
Fix plugin link tooltip flickering

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

The PR #10004 for "Eliminate unnecessary tooltip divs added for links" introduced a few issues when hovering over a  plugin `LinkTooltip` link:

- In certain mouse and screen positions, the tooltip be placed right at the pointer of the mouse, causing sporadic `mouseon` and `mouseleave` events to fire. This makes the tooltip flicker continuously.
- The protocol for this plugin feature has changed. Before the PR, the `LinkTooltip` component was mounted on initial render of the post containing the link. The component receives a `show` prop when it needs to react to being hovered. After the PR, the component is mounted on hover, making it so the component cannot do any local state caching.
- The popover does not fade into appearance anymore. It just instantly shows instead.

The first two bullets together is causing a spam of requests to fetch data for the plugin's component.

The changes in the linked PR are a little complicated, and I'm not sure which pieces are causing which issues, so I've decided to revert the PR.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-43906

#### Related Pull Requests

#10004

#### Release Note

```release-note
Fix plugin link tooltip flickering issue
```

### CODE REVIEW

File: components/link_tooltip/link_tooltip.tsx [hunk 1]
```diff
@@ -75,24 +74,17 @@ export default class LinkTooltip extends React.PureComponent<Props, State> {
                         }
                     });
                 };
-                this.tooltipContainer.childNodes.forEach(addChildEventListeners);
+                tooltipContainer.childNodes.forEach(addChildEventListeners);
 
-                this.popper = new Popper(this.target, this.tooltipContainer, {
-                    placement: 'auto',
+                this.popper = new Popper(target, tooltipContainer, {
+                    placement: 'bottom',
```
Comment (koox00) [inline components/link_tooltip/link_tooltip.tsx:80]: Is this change intended?

**Review by koox00 — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Reverting this sounds good for the time being, but I'd still like to address the issue of having Portals/Poppers created and components being mounted for every link posted as soon as one plugin might use them. In a channel with a lot of posts containing links, this adds a lot of invisible components that might never render anything. I can't see a way of doing that fully without changing the lifecycle of the tooltip component like this PR did.

Could we maybe reopen the original ticket and have your team look at it at some point in the future? That way, it's less likely that we end up making changes that mess up plugins in ways we're not aware of.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Ensured the flicker no longer reproduces
- Tested on browser and desktop app
- No issues found and relevant JS errors
- Ran E2E tests - No failures relevant to front end webapp changes

LGTM!
